R/DIFFUSE.R

```diff
@@ -0,0 +1,266 @@
+               iElement <- jTargetList[m] #retrieve name of target of node j
+
+               #W [i (receiver) , j (source)] is equal to 1/degree(j)
+               W[iElement,jElement] <- 1/vertDegree[names(vertDegree) == jElement]
+
+               doneEdges <- doneEdges + 1
+
+               consoleVect <- paste("Have value 1/deg(j) for",
+                                                   doneEdges,
+                                                   "of", sizeEdges,
+                                                   "assigned to W", sep = " ")
+               if (!silent) {
+                   print(consoleVect)
+               }
+
+           }
+
+        } #end construction of W
+
+    #======== Part 3 of Leis: calculating the matrix F ==============
+        betaVal <- param[[1]] #set beta Value, or the 'insulation' coefficient
+
+        #Following equation F= B*inv(I-(1-B)*W)
+
+        consoleVect <- "calculating matrix F"
+        if (!silent) {
+            print(consoleVect)
+        }
+
+        idMat <- diag(nrow = sizeVerts, ncol = sizeVerts)
+        preInvert <- idMat-(1-betaVal)*W #perform initial operations
+        postInvert <- solve(preInvert) #invert matrix
+        matrixF <- betaVal*postInvert #Finish and make matrix F
+
+    #======== Part 4 of Leis: calculating matrix E ===================
+
+        consoleVect <- c(consoleVect, "calculating matrix E")
+
+
+        #Make a diagonal matrix D, or vector of heat scores in diagonal form
+
+        D <- diag(x = AGGverts$Gene_Score)
+        dimnames(D) <- list(AGGverts$name,AGGverts$name)
+
+        #Calculate matrix E (weighted diffusion matrix) as F*D (F postmultiplied
+        #by D)
+
+        matrixE <- matrixF%*%D
+
+    #======== Part 5 of Leis: extracting 'heat' influence from matrix E =====
+
+        consoleVect <- c(consoleVect, "Assigning heat/influence scores from E to edges")
+
+
+        influenceVect <- numeric(length = sizeEdges)
+        #vector of zeros, to be filled with E(i,j) values
+
+        for (k in 1:sizeEdges) {
+            j <- AGGedges$from[k]
+            i <- AGGedges$to[k]
+
+            influenceVect[k] <- matrixE[i,j]
+        }
+
+        AGGedges$Influence <- influenceVect
+
+        EGG <- igraph::graph_from_data_frame(AGGedges, directed = TRUE,
+                                             vertices = AGGverts)
+
+    }
+    else {
+        stop("Error: No method selected")
+    }
+
+           finish <- Sys.time()
+
+        #setup metadata
+           meta <- list(version = "EGG_Version_1.0",
+                        UUID = uuid::UUIDgenerate(),
+                        input = paste("AGGuuID",metaAGG["UUID"][[1]], sep = ""),
+                        time = finish)
+          attr(EGG,"meta") <- meta
+#==================write the log=====================
+
+    if (!noLog) {
+        consoleVect <- "Writing Log"
+        if (!silent) {
+            print(consoleVect[length(consoleVect)])
```

Comment:
A: you've just overwritten your entire consoleVect with one string.
B: ```consoleVect[length(consoleVect)]``` prints only the last element.
C: that's the same as printing only the one string because of A.